src/components/VerboseTextField.tsx

```diff
@@ -31,6 +31,7 @@ import FieldNameLabel from "./FieldNameLabel";
 export interface VerboseTextFieldProps
   extends Omit<OutlinedTextFieldProps, "variant"> {
   state?: VerboseFieldState;
+  necessity?: "required" | "recommended";
```

Comment:
This type is duplicated, and could diverge between the two places. It would be better to have a single type definition.

Also, I think the type could be defined as follows: 
```
const necessityLevels = ["required" as const, "recommended" as const];
type NecessityType = typeof necessityLevels extends Array<infer E> ? E : never;
```
This way, you can do exchaustive checks and get compile errors when adding a new value, to make sure we don't forget to handle it: 
```
const printNecessity = (necessity: NecessityType) => {
  switch (necessity) {
    case "recommended":
      console.log("Recommended");
      return;
    case "required":
      console.log("Required");
      return;
    default:
      // eslint-disable-next-line no-case-declarations, no-underscore-dangle
      const _exhaustiveCheck: never = necessity;
      throw new Error(`unknown necessity: ${_exhaustiveCheck}`);
  }
};
```
If you add a third necessity level, you get an error in the switch default condition at compile time.

Reply:
That looks pretty cool!

Would there be any downsides in using the following though?

```typescript
export type Necessity = "required" | "recommended";

const necessityToLabel: Record<Necessity, string> = {
  recommended: "Recommended",
  required: "Required",
};
```
Adding another type here will complain as well.

Reply:
Sure, that works too!